api/data_refinery_api/views.py

```diff
@@ -540,7 +541,12 @@ def perform_update(self, serializer):
             accessions = new_data['data'][key]
             if accessions == ["ALL"]:
                 experiment = get_object_or_404(Experiment, accession_code=key)
-                sample_codes = list(experiment.samples.filter(is_processed=True).values_list('accession_code', flat=True))
+
+                # QN organisms
+                organism_ids = list(ComputationalResultAnnotation.objects.filter(data__is_qn=True).values_list('data__organism_id', flat=True).order_by('data__organism_id'))
+                organisms = Organism.objects.filter(id__in=organism_ids)
+
+                sample_codes = list(experiment.samples.filter(is_processed=True, organism__in=organisms).values_list('accession_code', flat=True))
```

Comment:
Should we do the same check for when someone wants to add a sample with an accession code? And throw an error if the sample doesn't have any QN targets associated?

Reply:
That's an interesting question but I'm not sure it's the API's duty to do that at that stage if we're returning them those objects only to have using them be a source of failure..